numba/cuda/compiler.py

```diff
@@ -260,8 +260,13 @@ def compile(self, args):
         Returns the `CompileResult`.
         """
         if args not in self._compileinfos:
+            nvvm_options = {
+                'opt': 3 if self.opt else 0,
+                'debug': self.debug,
+            }
+
             cres = compile_cuda(self.py_func, None, args, debug=self.debug,
-                                inline=self.inline)
+                                inline=self.inline, nvvm_options=nvvm_options)
```

Comment:
Whilst this replicates the `opt` behaviour in `inspect_ptx` is this now changing the default optimisation behaviour for callers of this method? Was it just being ignored before/something else would override it (direct inlining)?

Reply:
It's fixing the optimization behaviour - even if you passed `opt=False`, it would still optimize because we passed nothing to NVVM and it optimizes by default. Worse, the test case that checked this (`numba.cuda.tests.cudapy.test_optimization.TestOptimization.test_device_noopt`) was erroneously passing because of the incorrect implementation of `inspect_ptx`! Once I fixed the behaviour of `inspect_ptx()`, this test started failing, highlighting the need for the change on the line these comments pertain to.

Reply:
Thanks for the explanation. Seems like this is doing the correct behaviour now!